packages/roosterjs-editor-dom/lib/edit/adjustInsertPosition.ts

```diff
@@ -160,28 +159,11 @@ function adjustInsertPositionForStructuredNode(
                     : PositionType.After
             );
         }
-    } else if (tag == 'TABLE' && trNode) {
-        // When inserting a table into a table, if these tables have the same column count, and
-        // current position is at beginning of a row, then merge these two tables
-        let newTable = new VTable(<HTMLTableElement>rootNodeToInsert);
-        let currentTable = new VTable(<HTMLTableCellElement>tdNode);
+    }
 
-        if (
-            currentTable.col == 0 &&
-            tdNode == currentTable.getCell(currentTable.row || 0, 0).td &&
-            newTable.cells?.[0] &&
-            newTable.cells[0].length == currentTable.cells?.[0].length &&
-            isPositionAtBeginningOf(position, tdNode)
-        ) {
-            if (
-                getTagOfNode(rootNodeToInsert!.firstChild) == 'TBODY' &&
-                !rootNodeToInsert!.firstChild?.nextSibling
-            ) {
-                unwrap(rootNodeToInsert!.firstChild!);
-            }
-            unwrap(rootNodeToInsert!);
-            position = new Position(trNode, PositionType.After);
-        }
+    if (tag == 'TABLE' && trNode) {
+        pasteTable(root, nodeToInsert, position, range);
```

Comment:
you can pass in rootNodeToInsert so no need to check if it is TABLE again in pasteTable()